app/src/main/java/com/github/wnder/picture/Picture.java

```diff
@@ -102,4 +103,20 @@ public void onUpdatedGuessesAvailable(Consumer<Map<String, Location>> guessesAva
             guessesAvailable.accept(convertedResult);
         });
     }
+
+    /**
+     * Apply consumer function when the karma is available
+     * @param karmaAvailable consumer function to call when the karma is available
+     */
+    public void onKarmaUpdated(Consumer<Map<String, Object>> karmaAvailable){
+        Task<DocumentSnapshot> karmaTask = Storage.downloadFromFirestore("pictures", uniqueId);
+        karmaTask.addOnSuccessListener((documentSnapshot) -> {
+            Map<String, Object> map = new HashMap<>();
+            //We put these attributes back because if we don't, they disappear from the db
+            map.put("latitude", documentSnapshot.getDouble("latitude"));
+            map.put("longitude", documentSnapshot.getDouble("longitude"));
```

Comment:
There might be a way to do this with firestore (add/update fields without deleting the others).